src/librbd/migration/HttpClient.cc

```diff
@@ -607,12 +606,13 @@ class HttpClient<I>::PlainHttpSession : public HttpSession<PlainHttpSession> {
     auto cct = http_client->m_cct;
     ldout(cct, 15) << dendl;
 
-    ceph_assert(!m_stream.socket().is_open());
-    m_stream.async_connect(
-      results,
-      [on_finish](boost::system::error_code ec, const auto& endpoint) {
-        on_finish->complete(-ec.value());
-      });
+    ceph_assert(!m_stream.is_open());
+    boost::asio::async_connect(m_stream,
+			       results,
+			       [on_finish](boost::system::error_code ec,
+					   const auto endpoint) {
```

Comment:
`endpoint` was a const reference before (and still is in `SslHttpSession`):
```suggestion
					   const auto& endpoint) {
```